planning/collision_checker/include/collision_checker/node.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COLLISION_CHECKER__NODE_HPP_
+#define COLLISION_CHECKER__NODE_HPP_
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <tier4_autoware_utils/tier4_autoware_utils.hpp>
+#include <vehicle_info_util/vehicle_info_util.hpp>
+
+#include <autoware_adapi_v1_msgs/msg/operation_mode_state.hpp>
+#include <autoware_auto_perception_msgs/msg/predicted_objects.hpp>
+#include <autoware_auto_planning_msgs/msg/trajectory.hpp>
+#include <nav_msgs/msg/odometry.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+
+#include <tf2/utils.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
+#include <memory>
+#include <string>
+#include <utility>
+
+namespace collision_checker
+{
+using autoware_adapi_v1_msgs::msg::OperationModeState;
+using autoware_auto_perception_msgs::msg::PredictedObjects;
+using autoware_auto_perception_msgs::msg::Shape;
+using vehicle_info_util::VehicleInfo;
+
+using Obstacle = std::pair<double /* distance */, geometry_msgs::msg::Point>;
+
+class CollisionCheckerNode : public rclcpp::Node
+{
+public:
+  explicit CollisionCheckerNode(const rclcpp::NodeOptions & node_options);
+
+  struct NodeParam
+  {
+    bool use_pointcloud;
+    bool use_dynamic_object;
+    double collision_distance;
+  };
+
+private:
+  void checkCollision(diagnostic_updater::DiagnosticStatusWrapper & stat);
+
+  void onPointCloud(const sensor_msgs::msg::PointCloud2::ConstSharedPtr msg);
+
+  void onDynamicObjects(const PredictedObjects::ConstSharedPtr msg);
+
+  void onOperationMode(const OperationModeState::ConstSharedPtr msg);
+
+  // void onOdometry(const nav_msgs::msg::Odometry::ConstSharedPtr msg);
```

Comment:
Please remove the unnecessary comments.

Reply:
https://github.com/tier4/autoware.universe/pull/332/commits/a2d42bf5eee6d410b683716284cfd3c962f7f255